R/mz_summary.R

```diff
@@ -0,0 +1,185 @@
+#' The default is filter = F, so the unfiltered summary will be returned. 
+#' If filter is set to a `number`, e.g. 1000 only mz differences above 
+#' this threshold will be displayed. 
+#' The function can be applied for adjacency lists from `structural` and `combine`
+#' 
+#' @return 
+#' `data.frame` containing the numbers of present mz differences and
+#' corresponding name. 
+#'
+#' @author Liesa Salzer, \email{liesa.salzer@@helmholtz-muenchen.de}
+#'
+#' @examples
+#' data("x_test", package = "MetNet")
+#' transformation <- rbind(
+#'     c("Monosaccharide (-H2O)", "C6H10O5", "162.0528234315"),
+#'     c("Disaccharide (-H2O)", "C12H20O11", "340.1005614851"),
+#'     c("Trisaccharide (-H2O)", "C18H30O15", "486.1584702945"))
+#' transformation <- data.frame(group = transformation[, 1],
+#'                                 formula = transformation[, 2],
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' mz_summary(am_struct)
+#'
+#'
+#'@export
+#'
+#'@importFrom dplyr %>%
+mz_summary <- function(x, filter = FALSE, ...){
+  
+  if (!is(x, "AdjacencyMatrix")) {
+    stop("'x' is not an 'AdjacencyMatrix' object")
+  }
+  
+  if (!validObject(x)) 
+    stop("'x' must be a valid 'AdjacencyMatrix' object") 
+  
+  x_df <- as.data.frame(x)
+  
+  if (!"combine" %in% type(x) & !"structural" %in% type(x) ) 
+    stop("'x' does not contain a mass-difference adjacency matrix")
+  
+  if (!1 %in% x_df$binary) 
+    stop("'x' does not contain any mass-differences")
+  
+  if (!is.numeric(filter) & !FALSE %in% filter) 
+    stop("'filter' needs to be numeric or 'FALSE'")
+  
+  # if AdjacencyMatrix from `combine` is used 
+  if("combine" %in% type(x)){
+    x_df <- x_df[x_df$combine_binary == 1, ]
+    x_df <- na.omit(x_df, "combine_binary")
+    
+    sum_mass <- x_df %>% 
+      dplyr::group_by(combine_mass_difference) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- x_df %>% 
+      dplyr::group_by(combine_transformation) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$combine_mass_difference)
+  }
+  
+  # if AdjacencyMatrix from `structural` is used 
+  else {
+    x_df <- x_df[x_df$binary == 1, ]
+    
+    sum_mass <- x_df %>% 
+      dplyr::group_by(mass_difference) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- x_df %>% 
+      dplyr::group_by(transformation) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$mass_difference)
+  }
+  
+  colnames(sum_transform) <- c("transformation", "counts", "mass_difference")
+  sum_transform <- sum_transform %>% 
+    dplyr::select(transformation, mass_difference, counts)
+  
+  if (filter == FALSE | filter <= 1) {
```

Comment:
I guess if you use only `0`/positive integer for `filter` then the `if/else` statement is not needed and code simplifies to 

    sum_f <- filter(sum_transform,sum_transform$counts >= filter)
    return(sum_f)